ui/guest/src/utils/dom.ts

```diff
@@ -400,11 +393,68 @@ export function updateDropZoneValidations(
 }
 
 export function getZoneMarkerStyle(rect: DOMRect, padding: number = 0): CSSProperties {
-  const $window = $(window);
   return {
     height: rect.height + padding,
     width: rect.width + padding,
-    top: rect.top + $window.scrollTop() - padding / 2,
-    left: rect.left + $window.scrollLeft() - padding / 2
+    top: rect.top + window.scrollY - padding / 2,
+    left: rect.left + window.scrollX - padding / 2
   };
 }
+
+export function fadeIn(el: HTMLElement) {
+  var op = 0;
+  el.style.opacity = `${op}`;
+  el.style.display = 'inline-block';
+
+  var timer = setInterval(function () {
+    if (op >= 1.0) {
+      clearInterval(timer);
+    }
+    el.style.opacity = `${op}`;
+    op = op + 0.1;
+  }, 50);
+}
+
+export function elementOffset(element: Element) {
+  // Position of element relative to document = window scrolling position + position of element relative to screen
+  var rec = element.getBoundingClientRect();
+  var top = rec.top + window.scrollY;
+  var left = rec.left + window.scrollX;
+  return { top: top, left: left };
+}
+
+/* Traverse DOM from event target up to parent, searching for selector */
+function searchSelector(event, selector, stopAt) {
+  const nodes = [];
+  let currentNode = event.target;
+  let iterate = true;
+  const propagateEvents = ['dragover'];
+
+  while (iterate) {
+    if (currentNode.matches?.(selector)) {
+      nodes.push(currentNode);
+      currentNode = currentNode.parentNode;
+      if (!propagateEvents.includes(event.type)) {
+        iterate = false;
+      }
+    } else if (currentNode !== stopAt && currentNode !== document.body) {
+      currentNode = currentNode.parentNode;
+    } else {
+      iterate = false;
+    }
+  }
+  return nodes;
+}
+
+export function delegateEventListener(element, eName, selector, fn) {
+  element.addEventListener(eName, function (event) {
+    const nodes = searchSelector(event, selector, event.currentTarget);
```

Comment:
Not a great name. Also not used for anything else. Let's just merge the code here.